prody/dynamics/signature.py

```diff
@@ -901,8 +901,50 @@ def calcSignatureCrossCorr(mode_ensemble, norm=True):
         
     return sig
 
+def calcSignatureOverlaps(mode_ensemble):
+    """Calculate average mode-mode overlaps for a modeEnsemble (a list of modes)."""
+    
+    if not isinstance(mode_ensemble, ModeEnsemble):
+        raise TypeError('mode_ensemble should be an instance of ModeEnsemble')
+
+    if not mode_ensemble.isMatched():
+        LOGGER.warn('modes in mode_ensemble did not match cross modesets. '
+                    'Consider running mode_ensemble.match() prior to using this function')
+
+    n_sets = mode_ensemble.numModeSets()
+    n_modes = mode_ensemble.numModes()
+
+    overlaps = sdarray(array = np.zeros((n_sets,n_sets,n_modes,n_modes)))
+
+    for i, modeset_i in enumerate(mode_ensemble):
+        for j, modeset_j in enumerate(mode_ensemble):
+            overlaps[i,j] = abs(calcOverlap(modeset_i, modeset_j))
+
+    return overlaps
+
+def showSignatureOverlapsDiag(mode_ensemble):
+
+    if not isinstance(mode_ensemble, ModeEnsemble):
+        raise TypeError('mode_ensemble should be an instance of ModeEnsemble')
+
+    if not mode_ensemble.isMatched():
+        LOGGER.warn('modes in mode_ensemble did not match cross modesets. '
+                    'Consider running mode_ensemble.match() prior to using this function')
+
+    n_sets = mode_ensemble.numModeSets()
+    n_modes = mode_ensemble.numModes()
+
+    overlaps = calcSignatureOverlaps(mode_ensemble)
+
+    meanV = np.diag(overlaps.mean().mean(axis=0))
+    stdV = np.diag(overlaps.std().std(axis=0))
+    maxV = np.diag(overlaps.max().max(axis=0))
+    minV = np.diag(overlaps.min().min(axis=0))
+
+    return _showSignature(meanV, stdV, minV, maxV)
```

Comment:
I think you can just call ``showSignature`` here and ``_showSignature`` can remain to be a internal function of ``showSignature``. ``showSignature`` is designed to serve as what you are using ``_showSignature`` for.

Reply:
Ok I see.